From 92d66262c7bc83090e3716cc6b63df258590e764 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Mon, 4 Jul 2016 18:13:57 +0800
Subject: [PATCH] Revert "net: stmmac: replace msleep with mdelay between
 spinlock and spinunlock"

This reverts commit b664a51a074b560e92e5b22523c5cb54833e7bf2.
Upstream commit f55d84b07c4e ("stmmac: do not sleep in atomic context for mdio_reset")
fixes this bug.

Change-Id: I7332b3586640667f551c7e83eafff560a4f5a478
Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c
index 13af780ca8fc..bba670c42e37 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c
@@ -161,15 +161,15 @@ int stmmac_mdio_reset(struct mii_bus *bus)
 		gpio_direction_output(data->reset_gpio,
 				      data->active_low ? 1 : 0);
 		if (data->delays[0])
-			mdelay(DIV_ROUND_UP(data->delays[0], 1000));
+			msleep(DIV_ROUND_UP(data->delays[0], 1000));
 
 		gpio_set_value(data->reset_gpio, data->active_low ? 0 : 1);
 		if (data->delays[1])
-			mdelay(DIV_ROUND_UP(data->delays[1], 1000));
+			msleep(DIV_ROUND_UP(data->delays[1], 1000));
 
 		gpio_set_value(data->reset_gpio, data->active_low ? 1 : 0);
 		if (data->delays[2])
-			mdelay(DIV_ROUND_UP(data->delays[2], 1000));
+			msleep(DIV_ROUND_UP(data->delays[2], 1000));
 	}
 #endif
 
-- 
2.35.3

